test/unit/test_kinetic_linear.cpp

```diff
@@ -88,89 +89,111 @@ void run_test(std::string mech_name,
 
-    run_test<gpu::backend>("test0_kin_compartment", state_variables, {}, t0_values, t1_0_values);
-    run_test<gpu::backend>("test1_kin_compartment", state_variables, {}, t0_values, t1_1_values);
+    run_test<gpu::backend>("test0_kin_compartment", state_variables, {}, t0_values, t1_0_values, 0.5);
+    run_test<gpu::backend>("test1_kin_compartment", state_variables, {}, t0_values, t1_1_values, 0.5);
 }
 
-TEST(mech_kinetic_gpu, kintetic_1_conserve) {
+TEST(mech_kinetic_gpu, kintetic_linear_1_conserve) {
     std::vector<std::string> state_variables = {"s", "h", "d"};
     std::vector<fvm_value_type> t0_values = {0.5, 0.2, 0.3};
     std::vector<fvm_value_type> t1_0_values = {0.380338, 0.446414, 0.173247};
     std::vector<fvm_value_type> t1_1_values = {0.218978, 0.729927, 0.0510949};
 
-    run_test<gpu::backend>("test0_kin_diff", state_variables, {}, t0_values, t1_0_values);
-    run_test<gpu::backend>("test0_kin_conserve", state_variables, {}, t0_values, t1_0_values);
-    run_test<gpu::backend>("test0_kin_steadystate", state_variables, {}, t0_values, t1_1_values);
+    run_test<gpu::backend>("test0_kin_diff", state_variables, {}, t0_values, t1_0_values, 0.5);
+    run_test<gpu::backend>("test0_kin_conserve", state_variables, {}, t0_values, t1_0_values, 0.5);
+    run_test<gpu::backend>("test0_kin_steadystate", state_variables, {}, t0_values, t1_1_values, 0.5);
 }
 
-TEST(mech_kinetic_gpu, kintetic_2_conserve) {
+TEST(mech_kinetic_gpu, kintetic_linear_2_conserve) {
     std::vector<std::string> state_variables = {"a", "b", "x", "y"};
     std::vector<fvm_value_type> t0_values = {0.2, 0.8, 0.6, 0.4};
     std::vector<fvm_value_type> t1_0_values = {0.217391304, 0.782608696, 0.33333333, 0.66666666};
     std::vector<fvm_value_type> t1_1_values = {0.230769, 0.769231, 0.189189, 0.810811};
 
-    run_test<gpu::backend>("test1_kin_diff", state_variables, {}, t0_values, t1_0_values);
-    run_test<gpu::backend>("test1_kin_conserve", state_variables, {}, t0_values, t1_0_values);
-    run_test<gpu    ::backend>("test1_kin_steadystate", state_variables, {}, t0_values, t1_1_values);
+    run_test<gpu::backend>("test1_kin_diff", state_variables, {}, t0_values, t1_0_values, 0.5);
+    run_test<gpu::backend>("test1_kin_conserve", state_variables, {}, t0_values, t1_0_values, 0.5);
+    run_test<gpu::backend>("test1_kin_steadystate", state_variables, {}, t0_values, t1_1_values, 0.5);
+}
+
+TEST(mech_kinetic, kintetic_nonlinear) {
+    std::vector<std::string> state_variables = {"a", "b", "c"};
+    std::vector<fvm_value_type> t0_values = {0.2, 0.3, 0.5};
+    std::vector<fvm_value_type> t1_0_values = {0.222881, 0.31144, 0.48856};
+    std::vector<fvm_value_type> t1_1_values = {0.2078873133, 0.34222075, 0.45777925};
+
+    run_test<gpu::backend>("test2_kin_diff", state_variables, {}, t0_values, t1_0_values, 0.025);
+    run_test<gpu::backend>("test3_kin_diff", state_variables, {}, t0_values, t1_1_values, 0.025);
 }
 
```

Comment:
Again, given the grief from COMPARTMENT scaling up to now, can you add a test for a kinetic scheme with two different, interacting, COMPARTMENTs?

Reply:
Yes, I think that's a good idea, I will add a new test